test/unit/alphabet/nucleotide/dna15_test.cpp

```diff
@@ -15,60 +15,60 @@ INSTANTIATE_TYPED_TEST_CASE_P(dna15, alphabet, dna15);
 INSTANTIATE_TYPED_TEST_CASE_P(dna15, alphabet_constexpr, dna15);
 INSTANTIATE_TYPED_TEST_CASE_P(dna15, nucleotide, dna15);
 
-TEST(dna15, assign_char)
+TEST(dna15, to_char_assign_char)
 {
-    using t = dna15;
-    std::vector<char> chars
-    {
-        'A', 'C', 'G', 'T', 'U', 'N',
-        'a', 'c', 'g', 't', 'u', 'n',
-        'R', 'Y', 'S', 'W', 'K', 'M', 'B', 'D', 'H', 'V',
-        'r', 'y', 's', 'w', 'k', 'm', 'b', 'd', 'h', 'v',
-        '!'
-    };
+    EXPECT_EQ(to_char(dna15{}.assign_char('A')), 'A');
+    EXPECT_EQ(to_char(dna15{}.assign_char('C')), 'C');
+    EXPECT_EQ(to_char(dna15{}.assign_char('G')), 'G');
 
-    std::vector<dna15> alphabets
-    {
-        t::A, t::C, t::G, t::T, t::U, t::N,
-        t::A, t::C, t::G, t::T, t::U, t::N,
-        t::R, t::Y, t::S, t::W, t::K, t::M, t::B, t::D, t::H, t::V,
-        t::R, t::Y, t::S, t::W, t::K, t::M, t::B, t::D, t::H, t::V,
-        t::N
-    };
+    EXPECT_EQ(to_char(dna15{}.assign_char('U')), 'T');
+    EXPECT_EQ(to_char(dna15{}.assign_char('T')), 'T');
 
-    for (auto [ chr, alp ] : ranges::view::zip(chars, alphabets))
-        EXPECT_EQ((assign_char(dna15{}, chr)), alp);
+    EXPECT_EQ(to_char(dna15{}.assign_char('R')), 'R');
+    EXPECT_EQ(to_char(dna15{}.assign_char('Y')), 'Y');
+    EXPECT_EQ(to_char(dna15{}.assign_char('S')), 'S');
+    EXPECT_EQ(to_char(dna15{}.assign_char('W')), 'W');
+    EXPECT_EQ(to_char(dna15{}.assign_char('K')), 'K');
+    EXPECT_EQ(to_char(dna15{}.assign_char('M')), 'M');
+    EXPECT_EQ(to_char(dna15{}.assign_char('B')), 'B');
+    EXPECT_EQ(to_char(dna15{}.assign_char('D')), 'D');
+    EXPECT_EQ(to_char(dna15{}.assign_char('H')), 'H');
+    EXPECT_EQ(to_char(dna15{}.assign_char('V')), 'V');
+
+    EXPECT_EQ(to_char(dna15{}.assign_char('N')), 'N');
+    EXPECT_EQ(to_char(dna15{}.assign_char('!')), 'N');
 }
 
-TEST(dna15, to_char)
+TEST(dna15, char_literal)
 {
-    using t = dna15;
-    std::vector<char> chars
-    {
-        'A', 'C', 'G', 'T', 'T',
-        'R', 'Y', 'S', 'W', 'K', 'M', 'B', 'D', 'H', 'V',
-        'N'
-    };
+    EXPECT_EQ(to_char('A'_dna15), 'A');
```

Comment:
I think these files there the most annoying ones. Looks good!

Reply:
I personally would have tested if `'A'_dna15` is the same as `dna15{}.assign_rank(0)`, but this is completely fine.